sdk/storage/Azure.Storage.Common/src/Shared/SasExtensions.cs

```diff
@@ -260,27 +260,27 @@ internal static void AppendProperties(this SasQueryParameters parameters, String
 
             if (!string.IsNullOrWhiteSpace(parameters.CacheControl))
             {
-                stringBuilder.AppendQueryParameter(Constants.Sas.Parameters.CacheControl, parameters.CacheControl);
+                stringBuilder.AppendQueryParameter(Constants.Sas.Parameters.CacheControl, WebUtility.UrlEncode(parameters.CacheControl));
```

Comment:
Not a blocker but you could refactor this out into a method instead of rewriting the code a bunch of times.